src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+                        ref<uint8>(data_, 0) = LOGIN_ERROR_CHANGE_PASSWORD;
+                        do_write(1);
+                        return;
+                    }
+
+                    memset(data_, 0, 33);
+                    ref<uint8>(data_, 0) = LOGIN_SUCCESS_CHANGE_PASSWORD;
+                    do_write(33);
+
+                    ShowInfo("login_parse: password updated successfully.");
+                    return;
+                }
+            }
+            break;
+            default:
+            {
+                ShowError(fmt::format("Unhandled auth code: {}", code));
+            }
+            break;
+        }
+    }
+
+    void write_func() override
+    {
+        do_read();
+    }
+
+    void handle_error(std::error_code ec, std::shared_ptr<handler_session> self) override
+    {
+    }
+
+    void do_write(std::size_t length)
+    {
+        auto self(shared_from_this());
+        // clang-format off
+        asio::async_write(socket_, asio::buffer(data_, length),
+        [this, self](std::error_code ec, std::size_t /*length*/)
+        {
+            if (!ec)
+            {
+                write_func();
+            }
+            else
+            {
+                ShowError(ec.message());
+            }
+        });
+        // clang-format on
+    }
+};
+
+void PrintPacket(const char* data, uint32 size)
+{
+    std::string message;
+    char        buffer[5];
+
+    for (size_t y = 0; y < size; y++)
+    {
+        std::memset(buffer, 0, sizeof(buffer));               // TODO: Replace these three lines with std::format when/if we move to C++ 20.
+        snprintf(buffer, sizeof(buffer), "%02hhx ", data[y]); //
+        message.append(buffer);                               //
+
+        if (((y + 1) % 16) == 0)
+        {
+            message += "\n";
+            ShowDebug(message.c_str());
+            message.clear();
+        }
+    }
+
+    if (message.length() > 0)
+    {
+        message += "\n";
+        ShowDebug(message.c_str());
+    }
+}
+
+std::string getHashFromPacket(std::string ip_str, char* data)
+{
+    std::string hash = std::string(data + 12, 16);
+
+    if (authenticatedSessions_[ip_str].find(hash) == authenticatedSessions_[ip_str].end())
+    {
+        return "";
+    }
+    return hash;
+}
+
+// Main menu (Lobby), port 54001
+// A comment on the packets below, defined as macros.
+//   byte 0 - packet size
+//   bytes 4-7 are the packet header "IXFF" (0x49, 0x58, 0x46, 0x46)
+//   byte 8 - The expected message type:
+//     0x03 - positive result
+//     0x04 - error (in the case of an error, a uint16 error code is used at byte 32)
+//     Other
+class view_session : public handler_session
+{
+public:
+    view_session(asio::ssl::stream<asio::ip::tcp::socket> socket)
+    : handler_session(std::move(socket))
+    {
+        DebugSockets("view_session");
```

Comment:
In all of these debug prints, is it possible to also print out some information about the incoming connection/target socket?